core/src/main/java/org/elasticsearch/snapshots/SnapshotInfo.java

```diff
@@ -43,7 +45,7 @@
 /**
  * Information about a snapshot
  */
-public final class SnapshotInfo implements Comparable<SnapshotInfo>, ToXContent, Writeable {
+public final class SnapshotInfo implements Comparable<SnapshotInfo>, ToXContentObject, Writeable {
```

Comment:
This one is interesting and might be worth another look. To me it looks like this might write a fragment if this CONTEXT_MODE_PARAM parameter is set. Don't know what to do with it really, maybe mark for later?